### PR TITLE
DBZ-6878 add table specific notification in initial snapshot

### PR DESCRIPTION
**Pull request by vsantona**
DBZ-6878 add table specific notification in initial snapshot

### CODE REVIEW

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/NotificationsIT.java [hunk 1]
```diff
@@ -248,6 +267,37 @@ public void emittingDebeziumNotificationWillGenerateAJmxNotification()
...
+                .map(s -> ((Struct) s.value()))
+                .filter(v -> v.getString("type").equals("TABLE_SCAN_IN_PROGRESS") && v.getMap("additional_data").containsValue(tableName))
+                .findAny();
+        assertThat(tableNotification.isPresent()).isTrue();
+        assertThat(tableNotification.get().getString("aggregate_type")).isEqualTo("Initial Snapshot");
+        assertThat(tableNotification.get().getInt64("timestamp")).isCloseTo(Instant.now().toEpochMilli(), Percentage.withPercentage(1));
+
+        tableNotification = notifications.stream()
+                .map(s -> ((Struct) s.value()))
+                .filter(v -> v.getString("type").equals("TABLE_SCAN_COMPLETED") && v.getMap("additional_data").containsValue(tableName))
```
Comment (mfvitale) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/NotificationsIT.java:294]: Should `TABLE_SCAN_COMPLETED` notification tests also on `assertTableNotificationsSentToJmx`?

File: debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java [hunk 2]
```diff
@@ -581,7 +585,11 @@ private void doCreateDataEventsForTable(ChangeEventSourceContext sourceContext,
...
                     rows, table.id(), tableOrder, tableCount, Strings.duration(clock.currentTimeInMillis() - exportStart));
-            snapshotProgressListener.dataCollectionSnapshotCompleted(partition, table.id(), rows);
+            snapshotProgressListener.dataCollectionSnapshotCompleted(snapshotContext.partition, table.id(), rows);
+            LOGGER.info("notifyCompletedTable: " + table.id().table());
+            notificationService.initialSnapshotNotificationService().notifyCompletedTable(snapshotContext.partition,
+                    snapshotContext.offset,
+                    table.id().table());
         }
         catch (SQLException e) {
             throw new ConnectException("Snapshotting of table " + table.id() + " failed", e);
```
Comment (mfvitale) [inline debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:595]: Can be possible to add a status for `notifyCompletedTable` as for the incremental snapshot? Take a look to `TableScanCompletionStatus`.

File: debezium-core/src/main/java/io/debezium/pipeline/notification/InitialSnapshotNotificationService.java [hunk 3]
```diff
@@ -35,35 +36,52 @@ public InitialSnapshotNotificationService(NotificationService<P, O> notification
...
+                Map.of(SCANNED_COLLECTION, currentCollection)),
+                Offsets.of(partition, offsetContext));
+
+    }
+
+    public <T extends DataCollectionId> void notifyCompletedTable(P partition,
+                                                                  OffsetContext offsetContext,
+                                                                  String currentCollection) {
+        notificationService.notify(buildNotificationWith(SnapshotStatus.TABLE_SCAN_COMPLETED.name(),
+                Map.of(SCANNED_COLLECTION, currentCollection)),
```
Comment (mfvitale) [inline debezium-core/src/main/java/io/debezium/pipeline/notification/InitialSnapshotNotificationService.java:52]: It is possible to add information about the list of table to be scanned and also `total rows scanned` as for incremental snapshot notification?

**Comment by mfvitale** [comment]
Hi @vsantona , generally LGTM! I leaved just minor comments which would be better to fix before merge.

File: debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java (hunk 2 above)
Comment (vsantona) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:595]: Ok, I can add the status [SQL_EXCEPTION, SUCCEEDED]. Is this enough or do we also need the other states like EMPTY, SKIPPED, NO_PRIMARY_KEY and UNKNOWN_SCHEMA?

Comment (mfvitale) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:595]: I think that its fine to have `SQL_EXCEPTION, SUCCEEDED` for now.

**Comment by mfvitale** [comment]
@vsantona Applied, thanks!

**Pull request merged**